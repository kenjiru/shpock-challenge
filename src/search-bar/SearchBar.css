.page .search-bar {
    background: #e5e2dc;
    padding: 5px;
}

.page .search-bar-content {
    width: 1000px;
    margin: 0 auto;
    background: white;
}

.page .search-input {
    width: 384px;
    height: 20px;
    margin: 5px;
    padding: 2px 7px 3px;
    border: none;
    border-radius: 0;

    color: #707070;
    background: #fff;
    font: 15px/20px Arial, Helvetica, sans-serif;
    outline: none;
}
