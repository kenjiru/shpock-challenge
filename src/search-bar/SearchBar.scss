.search-bar {
    background: #e5e2dc;
    margin-bottom: 5px;

    .search-bar-content {
        padding: 5px;
        margin: 5px 0;
        background: white;

        .search-input {
            width: 100%;
            height: 20px;
            box-sizing: border-box;
            padding: 2px 7px 3px;
            border: none;
            border-radius: 0;

            color: #707070;
            background: #fff;
            font: 15px/20px Arial, Helvetica, sans-serif;
            outline: none;
        }
    }
}
