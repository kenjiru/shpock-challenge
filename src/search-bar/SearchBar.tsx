import * as React from "react";
import { Component, ReactElement } from "react";
import { Grid, Row, Col } from "react-flexbox-grid";

import "./SearchBar.scss";

class SearchBar extends Component<ISearchBarProps> {
    public static SEARCH_INPUT_CLASS: string = "search-input";
    private static PLACEHOLDER: string = "What are you looking for ...";

    public render(): ReactElement<HTMLElement> {
        return (
            <div className="search-bar">
                <Grid>
                    <Row>
                        <Col xs={0} md={2} lg={3}/>
                        <Col xs={12} md={8} lg={6}>
                            <div className="search-bar-content">
                                <input
                                    className={SearchBar.SEARCH_INPUT_CLASS}
                                    type="text"
                                    placeholder={SearchBar.PLACEHOLDER}
                                    value={this.props.searchStr}
                                    onChange={this.handleChange}
                                    onKeyDown={this.handleKeyDown}
                                />
                            </div>
                        </Col>
                        <Col xs={0} md={2} lg={3}/>
                    </Row>
                </Grid>
            </div>
        );
    }

    private handleKeyDown = (ev: React.KeyboardEvent<HTMLInputElement>): void => {
        if (ev.keyCode === 13) {
            ev.preventDefault();

            this.props.onSubmit();
        }
    }

    private handleChange = (ev: React.ChangeEvent<HTMLInputElement>): void => {
        this.props.onChange(ev.target.value);
    }
}

interface ISearchBarProps {
    searchStr: string;
    onSubmit: () => void;
    onChange: (searchStr: string) => void;
}

export default SearchBar;
