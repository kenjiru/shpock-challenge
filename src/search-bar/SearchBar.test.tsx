import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import shallowToJson from "enzyme-to-json";
import SearchBar from "./SearchBar";

describe("SearchBar", () => {
    it("should render correctly", () => {
        const fn = jest.fn();

        const output = shallow(
            <SearchBar searchStr="" onSubmit={fn} onChange={fn}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should call the onChange prop when input changes", () => {
        const INPUT_VALUE: string = "foo";
        const fn = jest.fn();
        const handleChange = jest.fn();

        const output: ShallowWrapper = shallow(
            <SearchBar searchStr="" onSubmit={fn} onChange={handleChange}/>
        );

        output.find("." + SearchBar.SEARCH_INPUT_CLASS).simulate("change", {
            target: {
                value: INPUT_VALUE
            }
        });

        expect(handleChange).toHaveBeenCalledWith(INPUT_VALUE);
    });

    it("should call the onSubmit prop when enter is pressed", () => {
        const fn = jest.fn();
        const handleSubmit = jest.fn();
        const preventDefaultSpy = jest.fn();

        const output: ShallowWrapper = shallow(
            <SearchBar searchStr="" onSubmit={handleSubmit} onChange={fn}/>
        );

        output.find("." + SearchBar.SEARCH_INPUT_CLASS).simulate("keyDown", {
            keyCode: 13,
            preventDefault: preventDefaultSpy
        });

        expect(handleSubmit).toHaveBeenCalled();
        expect(preventDefaultSpy).toHaveBeenCalled();
    });
});
