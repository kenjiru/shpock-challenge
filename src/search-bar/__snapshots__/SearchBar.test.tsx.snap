// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<div
  className="search-bar"
>
  <Grid>
    <Row>
      <Col
        lg={3}
        md={2}
        xs={0}
      />
      <Col
        lg={6}
        md={8}
        xs={12}
      >
        <div
          className="search-bar-content"
        >
          <input
            className="search-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="What are you looking for ..."
            type="text"
            value=""
          />
        </div>
      </Col>
      <Col
        lg={3}
        md={2}
        xs={0}
      />
    </Row>
  </Grid>
</div>
`;
